ctapipe/io/eventsource.py

```diff
@@ -200,6 +196,40 @@ def subarray(self):
 
         """
 
+    @property
+    @abstractmethod
+    def is_simulation(self):
+        """
+        Weither the currently opened file is simulated
+
+        Returns
+        -------
+        bool
+
+        """
+
+    @property
+    @abstractmethod
+    def datalevels(self):
+        """
+        The datalevels provided by this event source
+
+        Returns
+        -------
+        tuple[ctapipe.io.DataLevel]
+        """
+
+    @property
+    @abstractmethod
+    def obs_id(self):
+        """
```

Comment:
Eventually this should be a full ObservationConfiguration but since that is not well defined yet, it's ok.   That way, you get the InstrumentConfiguration (SubarrayDescription) and the ObservationConfiguration as "metadata" attached to the source.   The ObservationConfiguration would contain other parameters describing the observation mode, etc (like the mcheader for sims, which is.a sort of SimulatedObservationConfiguration), though of course they are all  keyed by the obs_id, so this is fine. 

In fact, we could make that generic: e.g. define that all EventSources provide an ObservationConfiguration (ObsConfigContainer?), and we just make MCHeader a sub-class of this, adding parameters specific to MCs.

Reply:
Note that you already get the obs_id inside the MCHeader (for simulations)

Reply:
Yes, this is where I get it in the SimTelEventSource